scripts/server.js

```diff
@@ -730,12 +730,12 @@ setups.getBackendSetups(function (setups) {
 
   // Start listening for requests.
   httpServer.listen(port, (err) => {
-    if (process.env.NODE_ENV === "development") {
+    if (process.env.NODE_ENV === "development") 
       setTimeout(setupDevServer, 2000);
-      app.get("/", ensureUser(), (req, res) => {
-        res.redirect(`http://localhost:${port + 1}`);
-      });
-    } else {
+    //   app.get("/", ensureUser(), (req, res) => {

```

Comment:
Removing this DEV vs PROD distinction is breaking hot-reloading in DEV (because PROD is solely relying on the /build folder and scripts)

Reply:
This change was not supposed to be in the pull, I modified it to run locally

Reply:
Could you also detail the steps you've been taking to run this locally? We are interested in simplifying the initial installation and supporting a variety setups. Often we develop simply with `npm start` and only use Docker for final production images if the customer wishes to go that route.

Reply:
Within the dockerfile, each time the FROM statement is included, a new layer of the image is created, so I divide the construction of the docker image with multi-stage.

### Normally
   - the bases
   - as builder
   - as develop
   - and final version

The big trick when running in development is to configure the volume, basically building a volume pointing to the src and in the script instead of starting with node, using nodemon for example, in java using Dev tools dependency, then it is possible to run in development.

And in the final version, I only copy the build / out file and only the run time in this image, unlike the dev one, here the intention is to leave just one image with instructions to run the package and as light as possible.

---
My sample of Dockerfile

```
########### Stage BUILD ###########
FROM maven:3.6.3-jdk-8-slim as builder

# Define Workdir
WORKDIR /opt/app
#Copy pom.xml to container
COPY pom.xml ./

RUN mvn dependency:go-offline

COPY ./src ./src
RUN mvn clean
RUN mvn install package

########### Stage RUN ###########
FROM eclipse-temurin:8-jre-alpine@sha256:f6485e494f9927cb6a51e23134b3556c62c32c3df20fc4317f7f493343726977 as runner
SHELL ["/bin/sh", "-c"]

RUN mkdir -p /etc/skel/

RUN echo 'history -c \
/bin/rm -f /opt/remote/.mysql_history \
/bin/rm -f /opt/remote/.history \
/bin/rm -f /opt/remote/.bash_history' >> /etc/skel/.logout

RUN echo 'set autologout = 30 \
set prompt = "$ " \
set history = 0 \
set ignoreeof' >> /etc/skel/.cshrc

RUN cp /etc/skel/.cshrc /etc/skel/.profile

# dumb-init occupies PID 1 and takes care of all the responsibilities.
RUN apk add dumb-init

# Define Workdir
WORKDIR /opt/app

# Add user javauser based minumum privileges
RUN addgroup -S -g 1000 javagroup && \
    adduser -S -s /bin/false -G javagroup -u 999 javauser

RUN echo "javauser:secretPass" | chpasswd

# Copy target from build stage
COPY --from=builder /opt/app/target/AuthService.war ./

RUN chmod 0 /home \
    && chmod 0 /var \
    && chmod 0 /usr \
    && chmod 0 /dev \
    && chmod 0 /mnt \
    && chmod go-rw /etc \
    && chmod 0 /etc/ssl \
    && chmod 0 /etc/skel \
    && chmod a+rwx /usr/bin/dumb-init

RUN chown -R javauser:javagroup /opt/app
USER javauser

ENTRYPOINT ["dumb-init", "java", "-jar", "AuthService.war"] # I will create an entrypoint

```

> In the case of this image presented above, I run it through compose pointing the target to the builder version and I configure a volume in the ./src folder